src/realm/group_shared.cpp

```diff
@@ -612,15 +612,14 @@ void spawn_daemon(const std::string& file)
         int i;
         for (i = m - 1; i >= 0; --i)
             close(i);
-        i = ::open("/dev/null", O_RDWR);
 #ifdef REALM_ENABLE_LOGFILE
         // FIXME: Do we want to always open the log file? Should it be configurable?
         i = ::open((file + ".log").c_str(), O_RDWR | O_CREAT | O_APPEND | O_SYNC, S_IRWXU);
 #else
-        i = dup(i);
+        i = ::open("/dev/null", O_RDWR);
 #endif
-        i = dup(i);
-        static_cast<void>(i);
+        int j = dup(i);
+        static_cast<void>(j);
```

Comment:
This code is functionally different than before. "/dev/null" is no longer open in case of `REALM_ENABLE_LOGFILE` I don't understand why we `dup` the files in the first place though if we never use them again. Do you know what's going on here? Does `dup` in a child process have some effect I am missing?